### ISSUE TITLE
Implementing word2vec using TF Transform

### ISSUE THREAD

**Issue by xennygrimmato** [tensorflow/transform#107]
**Usecase**:
- Given a PCollection of sentences, create word embeddings using word2vec (skip-gram model)

**Questions**:
1. Is there an existing Beam implementation that the developers and contributors of TF Transform can share that serves my purpose?
2. If not, is there some example that I can refer to so that I can implement it myself? Something that takes inputs from a PCollection and passes it on to a Tensorflow model.

I would like to understand thoroughly how I can operate with both Beam and Tensorflow.

Apologies in advance if these seem to be novice questions, but I would really like to implement this in the best possible way.

**Comment by hgadig** [comment]
Please confirm if you got a chance to go through the following links which has word2vec examples.

https://towardsdatascience.com/learn-word2vec-by-implementing-it-in-tensorflow-45641adaf2ac
https://www.tensorflow.org/tutorials/representation/word2vec

**Comment by xennygrimmato** [comment]
@hgadig Yes, I have gone through these.

**Comment by xennygrimmato** [comment]
@hgadig Please let me know if you need additional clarifications from my end about what I'm looking for.

I am aware of the implementation of word2vec using Tensorflow, but I don't know how to use PCollections with Tensorflow models.

**Comment by hgadig** [comment]
@zoyahav  Hi, could you PTAL at this.

**Comment by KesterTong** [comment]
There is no Beam implementation of word2vec training (that is, learning embeddings using the skip-gram model), and I would not recommend creating one.  Transform is intended for computations that are suited to the map-reduce model, while the skip-gram model is trained with gradient descent.

Instead, I suggest training word2vec with a standard TensorFlow model.

A pre-trained word2vec embedding can be applied with tensorflow_transform.
